### PR TITLE
Google login

### PR DESCRIPTION
**Pull request by aydevelop**

### CODE REVIEW

File: frontend/src/app/models/auth/auth-google-dto.ts [hunk 1]
```diff
@@ -0,0 +1 @@
+
```
Comment (shmato4ek) [inline frontend/src/app/models/auth/auth-google-dto.ts]: I think that this file is unnecessary

**Review by shmato4ek — COMMENTED** [review]
Check linter, there are warnings with it

File: frontend/src/app/models/auth/auth-google-dto.ts (hunk 1 above)
Comment (aydevelop) [reply frontend/src/app/models/auth/auth-google-dto.ts]: Fixed

**Review by shmato4ek — DISMISSED** [review]

File: backend/Recodo.API/Controllers/AuthController.cs [hunk 2]
```diff
@@ -43,5 +40,26 @@ public async Task<ActionResult<AuthUserDTO>> Login([FromBody] LoginUserDTO userD
...
+
+        [HttpPost("GoogleLogin")]
+        public async Task<IActionResult> GoogleLogin([FromBody] ExternalAuthDto externalAuth)
+        {
+            var payload = await _authService.VerifyGoogleToken(externalAuth);
+            if (payload == null)
+            {
+                return BadRequest("Invalid External Authentication.");
+            }
+
```
Comment (IvanitskyiVolodymyr) [inline backend/Recodo.API/Controllers/AuthController.cs:53]: Can you check payload inside the VerifyGoogleToken method, and if it is not valid, throw out new Exception like VerifyGoogleTokenException that will be handled by ErrorHandlerMiddleware

File: backend/Recodo.BLL/Services/AuthService.cs [hunk 3]
```diff
@@ -64,5 +70,23 @@ public async Task<TokenDTO> GenerateAccessToken(int id, string userName, string
             return new TokenDTO(accessToken, refreshToken);
         }
 
+        public async Task<GoogleJsonWebSignature.Payload> VerifyGoogleToken(ExternalAuthDto externalAuth)
+        {
+            try
+            {
+                string clientId = _configuration["GoogleClientId"];
+                var settings = new GoogleJsonWebSignature.ValidationSettings()
```
Comment (Jyuart) [inline backend/Recodo.BLL/Services/AuthService.cs:78]: Is it possible to configure it once on the start of the application (I suppose there should be an extension for AddAuthentication() in Startup class)

Comment (aydevelop) [reply backend/Recodo.BLL/Services/AuthService.cs:78]: It's just a class with a field "Audience" that is passed to ValidateAsync. It's not settings

File: backend/Recodo.API/Controllers/AuthController.cs (hunk 2 above)
Comment (aydevelop) [reply backend/Recodo.API/Controllers/AuthController.cs:53]: Done

**Review by Anatolii-Ivanov — APPROVED** [review]

**Review by Jyuart — APPROVED** [review]

**Pull request merged**